lib/network_utils.ml

```diff
@@ -487,7 +487,9 @@ module Linux_bonding = struct
 		if is_bond_device master then begin
 			let get_prop prop =
 				try
-					Some (prop, Sysfs.read_one_line (Sysfs.getpath master ("bonding/" ^ prop)))
+					if prop = "mode" then
+						Some (prop, List.hd (String.split ' ' (Sysfs.read_one_line (Sysfs.getpath master ("bonding/" ^ prop)))))
+					else Some (prop, Sysfs.read_one_line (Sysfs.getpath master ("bonding/" ^ prop)))
```

Comment:
I think `(Sysfs.read_one_line (Sysfs.getpath master ("bonding/" ^ prop)))` should be let-bound before the if-expression, which then becomes simpler. Overall this feels as if the format of some configuration file has changed in a way that wasn't anticipated and now the code here has to deal with it in an ad-hoc fashion.